eslint-plugin-expensify/tests/boolean-conditional-rendering.test.js

```diff
@@ -0,0 +1,260 @@
+const RuleTester = require('@typescript-eslint/rule-tester').RuleTester;
+const rule = require('../boolean-conditional-rendering');
+
+const ruleTester = new RuleTester({
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: './tsconfig.json',
+        tsconfigRootDir: __dirname,
+        sourceType: 'module',
+        ecmaVersion: 2020,
+        ecmaFeatures: {
+            jsx: true,
+        },
+    },
+});
+
+ruleTester.run('boolean-conditional-rendering', rule, {
```

Comment:
Can you please add a test for the boolean | undefined case we have seen a lot in the app?

Reply:
The test seems right to me and considering this is correctly flagging in App based on your testing, I am not sure. Asked Szymon in Slack

Reply:
But should we raise an error if the value is `boolean | undefined`? IMO we shouldn't, if the value is equal `undefined` the condition will be false.

Reply:
I think its better to be more strict when it comes to these than less, it should be easily resolvable by using !!